xbmc/guilib/GUIWindowManager.cpp

```diff
@@ -1247,6 +1248,33 @@ int CGUIWindowManager::GetActiveWindow() const
   return WINDOW_INVALID;
 }
 
+int CGUIWindowManager::GetActiveWindowID()
+{
+  // Get the currently active window
+  int iWin = GetActiveWindow() & WINDOW_ID_MASK;
+
+  // If there is a dialog active get the dialog id instead
+  if (HasModalDialog())
+    iWin = GetTopMostModalDialogID() & WINDOW_ID_MASK;
```

Comment:
we could use `CGUIWindowManager::GetFocusedWindow()` and get rid of the call to `GetActiveWindow()` and this conditional.

```
int iWin = GetFocusedWindow() & WINDOW_ID_MASK;
```